### PR TITLE
Query cleanup

### PR DESCRIPTION
**Pull request by smitpatel**
Most of changes are renames and moving files to Internal folder. I will comment on the breaking changes for visibility.

### CODE REVIEW

File: src/EFCore/Query/ExpressionPrinter.cs [hunk 1]
```diff
@@ -51,8 +51,6 @@ public ExpressionPrinter()
             _encounteredParameters = new List<ParameterExpression>();
         }
 
-        public virtual IndentedStringBuilder StringBuilder => _stringBuilder;
```
Comment (smitpatel) [inline src/EFCore/Query/ExpressionPrinter.cs]: Breaking change. Use API on expression printer directly

File: src/EFCore/Query/ExpressionPrinter.cs [hunk 2]
```diff
@@ -101,18 +99,6 @@ public virtual ExpressionPrinter AppendLines([NotNull] object o, bool skipFinalN
             return this;
         }
 
-        public virtual ExpressionPrinter IncrementIndent()
```
Comment (smitpatel) [inline src/EFCore/Query/ExpressionPrinter.cs]: Remove Increment/Decrement Indent. Using Indent and disposable pattern.

**Comment by smitpatel** [comment]
>     public class QueryOptimizerFactory : IQueryOptimizerFactory

Moved to Internal as it is factory.

---
Refers to: src/EFCore/Query/QueryOptimizerFactory.cs:18 in d309c50. [](commit_id = d309c503034fe19fc653bba0d03d3b6d55a434e1, deletion_comment = True)

**Comment by ajcvickers** [comment]
@smitpatel I intentionally made the factory public because we have dependencies for it, so it makes sense for providers to extend from our factory.

File: src/EFCore.Relational/Query/RelationalShapedQueryOptimizer.cs [hunk 3]
```diff
@@ -32,17 +35,19 @@ public override Expression Visit(Expression query)
...
             }
 
-            query = new SqlExpressionOptimizingVisitor(SqlExpressionFactory, UseRelationalNulls).Visit(query);
+            query = OptimizeSqlExpression(query);
             query = new NullComparisonTransformingExpressionVisitor().Visit(query);
 
             return query;
         }
+
+        protected virtual Expression OptimizeSqlExpression(Expression query) => _sqlExpressionOptimizingExpressionVisitor.Visit(query);
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalShapedQueryOptimizer.cs]: All visitors in optimizers are supposed to be internal and exposed via protected methods. This particular visitor moved to .Internal but being used by SqlServer so I have added method to facilitate this for now. We can go over and add methods to call other visitors too in future.

**Comment by smitpatel** [comment]
@ajcvickers - I will check and revert if necessary.

File: src/EFCore.Relational/Query/RelationalQueryContextFactory.cs [hunk 4]
```diff
@@ -38,15 +38,6 @@ public class RelationalQueryContextFactory : QueryContextFactory
         /// </summary>
         protected virtual RelationalQueryContextDependencies RelationalDependencies { get; }
 
-        /// <summary>
-        ///     The execution strategy factory.
-        /// </summary>
-        /// <value>
-        ///     The execution strategy factory.
-        /// </value>
-        protected virtual IExecutionStrategyFactory ExecutionStrategyFactory
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalQueryContextFactory.cs]: Factory should not be exposing anything from Dependency object.

File: src/EFCore/Query/IParameterValues.cs [hunk 5]
```diff
@@ -11,9 +11,5 @@ public interface IParameterValues
         IReadOnlyDictionary<string, object> ParameterValues { get; }
 
         void Add([NotNull] string name, [CanBeNull] object value);
-
-        object Remove([NotNull] string name);
```
Comment (smitpatel) [inline src/EFCore/Query/IParameterValues.cs]: These APIs were used in old pipeline. Not anymore. It should not affect any provider.

File: src/EFCore/Query/Internal/AnonymousObject.cs [hunk 6]
```diff
@@ -117,21 +81,5 @@ public override int GetHashCode()
...
             return hash.ToHashCode();
         }
-
-        /// <summary>
-        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
-        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
-        ///     any release. You should only use it directly in your code with extreme caution and knowing that
-        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
-        /// </summary>
-        public object GetValue(int index) => _values[index];
```
Comment (smitpatel) [inline src/EFCore/Query/Internal/AnonymousObject.cs]: This was to support include in old pipeline. No longer needed.

**Comment by smitpatel** [comment]
Added another commit to 
- Remove QueryContextFactory. (Was abstract class which did no specific computation).
- Providers implement interface directly. Relational factory was also moved to internal. This is provider breaking change. Npgsql should work since it derives from Relational.

**Comment by smitpatel** [comment]
We will go through breaking changes in API review again. 🎉

**Pull request merged**